### PR TITLE
feat: calculate flex consumption per grid area without grid loss

### PR DESCRIPTION
**Pull request by johevemi**
## Description

Added to result (without grid loss or system correction):
flex_consumption_ga
flex_consumption_ga_es
flex_consumption_ga_brp
production_ga_es
production_ga_brp

Also added to result:
grid_loss
added_grid_loss now positive_grid_loss
added_system_correction now  negative_grid_loss

Missing:
flex_consumption_ga_brp_es with grid loss
production _ga_brp_es with system correction

## References

* #907

### CODE REVIEW

File: source/databricks/calculation-engine/package/balance_fixing.py
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/package/balance_fixing.py:1]: Can we merge #1031 first and then update your PR with new test cases in `source/databricks/calculation-engine/tests/integration/calculator/calculator_job/test_calculator_job_results.py`? It should be very easy to do.

Comment (johevemi) [reply source/databricks/calculation-engine/package/balance_fixing.py:1]: Done

Comment (johevemi) [reply source/databricks/calculation-engine/package/balance_fixing.py:1]: nice test btw

Comment (BjarkeMeier) [reply source/databricks/calculation-engine/package/balance_fixing.py:1]: Thx 😄

File: source/databricks/calculation-engine/package/balance_fixing.py [hunk 1]
```diff
@@ -108,18 +223,62 @@ def _calculate_production(
...
-    enriched_time_series_point_df: DataFrame,
+    temporay_flex_consumption_per_per_ga_and_brp_and_es: DataFrame,
+    grid_loss: DataFrame,
 ) -> None:
-    # Non-profiled consumption per balance responsible party and energy supplier
-    consumption_per_ga_and_brp_and_es = (
-        agg_steps.aggregate_non_profiled_consumption_ga_brp_es(
-            enriched_time_series_point_df
-        )
+    added_grid_loss = agg_steps.calculate_added_grid_loss(grid_loss)
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/balance_fixing.py:231]: why is grid loss part of _calculate_flex_consumption? Would it make sense to have a separate function for this and then pass it as a parameter to flex?

File: source/databricks/calculation-engine/package/balance_fixing.py [hunk 2]
```diff
@@ -92,13 +122,98 @@ def _calculate_net_exchange_per_ga(
...
+    temporay_flex_consumption_per_per_ga_and_brp_and_es: DataFrame,
+    consumption_per_ga_and_brp_and_es: DataFrame,
+) -> DataFrame:
+    grid_loss = agg_steps.calculate_grid_loss(
+        net_exchange_per_ga,
+        temporay_production_per_per_ga_and_brp_and_es,
+        temporay_flex_consumption_per_per_ga_and_brp_and_es,
+        consumption_per_ga_and_brp_and_es,
+    )
+    result_writer.write(
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/balance_fixing.py:169]: are we missing some tests that verifies what is stored? Especially in the delta table.
